doc/source/tutorials/reading.rst

```diff
@@ -577,11 +592,93 @@ that your memory budget is reasonable, the above approach will complete
 quickly and any extra cost stemming from pausing and resuming the
 query gets amortized over the entire execution.
 
+Multi-range subarrays
+---------------------
+
+So far all the examples involved a single multi-dimensional rectangular range,
+created by specifying a single 1D range per dimension. TileDB also supports
+*multi-range* subarrays, i.e., it allows specifying multiple 1D ranges
+per dimension. The resulting subarray consists of the *cross product* of
+all the 1D ranges across all dimensions.
+
+The feature is better illustrated with an example. The figure below illustrates
+multi-range subarray ``{[1,2], [4,4]}, [1,4]``, which slices rows ``1, 2, 4``
+(two ranges along the rows dimension) and columns ``1, 2, 3, 4`` (one range
+along the columns dimension). For dense arrays, the permissible layouts are
+*row-major* and *column-major*. Note that the *global* layout is not supported
+for multi-range subarrays (whereas *unordered* reads are not supported at all
```

Comment:
I thought we were going to pick an arbitrary order for the dense case (or the slightly fancier least effort order depending on the layout)?